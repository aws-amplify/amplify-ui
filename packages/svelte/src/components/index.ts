export { default as Authenticator } from './Authenticator/Authenticator.svelte';
export { default as ConfirmResetPassword } from './Authenticator/ConfirmResetPassword.svelte';
export { default as ConfirmSignIn } from './Authenticator/ConfirmSignIn.svelte';
export { default as ConfirmSignUp } from './Authenticator/ConfirmSignUp.svelte';
export { default as FederatedSignIn } from './Authenticator/FederatedSignIn.svelte';
export { default as ForceNewPassword } from './Authenticator/ForceNewPassword.svelte';
export { default as ForgotPassword } from './Authenticator/ForgotPassword.svelte';
export { default as SignIn } from './Authenticator/SignIn.svelte'
export { default as SignUp } from './Authenticator/SignUp.svelte';
export { default as SetupTotp } from './Authenticator/SetupTotp.svelte';

export { default as AmplifyButton } from './primitives/Button.svelte'
export { default as AmplifyPasswordField } from './primitives/PasswordField.svelte'
export { default as AmplifyTextField } from './primitives/TextField.svelte'

export * from '../types';
