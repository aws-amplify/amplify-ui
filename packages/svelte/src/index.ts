/**
 * @aws-amplify/ui-svelte
 */

// Export version
export const VERSION = '0.0.1';

// Export stores and composables
export { useAuthenticator, useAuthenticatorStore } from './composables/useAuthenticator';
export { createAuthenticatorStore, stopAuthenticatorService, type UseAuthenticator } from './stores/authenticator';

// Export primitive components
export { Button, TextField, PasswordField } from './components/primitives';

// Export Authenticator components
export { Authenticator } from './components/Authenticator';

// Export types
export * from './types';