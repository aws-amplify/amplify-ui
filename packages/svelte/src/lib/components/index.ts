export * from './Authenticator';
export * from './controls';
export * from './primitives';

export * from '../types';
