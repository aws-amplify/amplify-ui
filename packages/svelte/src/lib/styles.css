@import '@aws-amplify/ui/styles.css';
@import './styles/authenticator.css';
