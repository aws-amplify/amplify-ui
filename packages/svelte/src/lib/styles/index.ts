// This file is intentionally empty
// CSS imports are handled directly in Svelte components
