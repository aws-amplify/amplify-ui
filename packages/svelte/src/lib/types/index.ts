import {
  AuthenticatorServiceFacade,
  AuthInterpreter,
  AuthMachineSend,
  AuthMachineState,
  AuthStatus,
} from '@aws-amplify/ui';
import type { Snippet } from 'svelte';
import type { Writable } from 'svelte/store';

export interface UseAuthenticator extends AuthenticatorServiceFacade {
  QRFields: { totpIssuer?: string; totpUsername?: string } | null;
}

export interface UseAuth {
  state: Writable<AuthMachineState>;
  send: AuthMachineSend;
  service: AuthInterpreter;
  authStatus: AuthStatus;
}

export interface Components {
  Footer?: Snippet;
  FormFields?: Snippet;
  Header?: Snippet;
}

//eslint-disable-next-line @typescript-eslint/no-explicit-any
export type AnyFn = (...args: any[]) => any;
