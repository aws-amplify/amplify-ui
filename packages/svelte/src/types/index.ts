/**
 * Re-export types from @aws-amplify/ui for convenience
 */
export type {
  AuthStatus,
  AuthUser,
  AuthenticatorRoute,
  ChallengeName,
  SocialProvider,
  AuthMFAType,
  UnverifiedUserAttributes,
  AuthenticatorValidationErrors,
  V5CodeDeliveryDetails,
  AuthEventData,
} from '@aws-amplify/ui';

/**
 * Svelte-specific component props
 */
export interface AuthenticatorProps {
  /**
   * Initial route to display
   */
  initialRoute?: AuthenticatorRoute;
  
  /**
   * Social providers to display
   */
  socialProviders?: SocialProvider[];
  
  /**
   * Custom theme
   */
  theme?: Record<string, any>;
  
  /**
   * Whether to hide the sign up link
   */
  hideSignUp?: boolean;
  
  /**
   * Custom form fields configuration
   */
  formFields?: Record<string, any>;
}

/**
 * Slot props passed to child components
 */
export interface AuthenticatorSlotProps {
  /**
   * Current authentication status
   */
  authStatus: AuthStatus;
  
  /**
   * Authenticated user
   */
  user: AuthUser | undefined;
  
  /**
   * Sign out function
   */
  signOut: () => void;
}

/**
 * Common props for authentication form components
 */
export interface AuthFormProps {
  /**
   * Error message to display
   */
  error?: string;
  
  /**
   * Whether the form is submitting
   */
  isPending?: boolean;
  
  /**
   * Validation errors
   */
  validationErrors?: Record<string, string>;
}