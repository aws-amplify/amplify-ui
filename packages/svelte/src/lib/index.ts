import Authenticator from './components/Authenticator.svelte';
import * as authStore from '$lib/components/authStore';

export { Authenticator, authStore };
