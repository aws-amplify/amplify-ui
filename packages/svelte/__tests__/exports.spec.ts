import { describe, expect, it } from 'vitest';

import * as exported from '../src/lib';

describe('@aws-amplify/ui-svelte', () => {
	describe('exports', () => {
		it('should match snapshot', () => {
			const sortedExports = Object.keys(exported).sort();
			expect(sortedExports).toMatchSnapshot();
		});
	});
});
