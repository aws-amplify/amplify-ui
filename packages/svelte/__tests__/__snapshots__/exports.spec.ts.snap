// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`@aws-amplify/ui-svelte > exports > should match snapshot 1`] = `
[
  "AliasField",
  "Authenticator",
  "Button",
  "Checkbox",
  "ConfirmResetPassword",
  "ConfirmSignIn",
  "ConfirmSignUp",
  "ConfirmVerifyUser",
  "FederatedSignIn",
  "ForceNewPassword",
  "ForceNewPasswordFormFields",
  "ForgotPassword",
  "PasswordField",
  "SelectMfaType",
  "SetupEmail",
  "SetupTotp",
  "SignIn",
  "SignUp",
  "SignUpFormFields",
  "TextField",
  "VerifyUser",
  "translations",
  "useAuthenticator",
]
`;
